pallets/dapp-staking/src/test/tests.rs

```diff
@@ -3487,3 +3789,363 @@ fn claim_bonus_reward_for_works() {
+            assert_eq!(final_staking_info, expected_final_staking_info);
+        })
+}
+
+#[test]
+fn move_stake_multiple_conversions_are_ok() {
+    ExtBuilder::default().build_and_execute(|| {
+        // Prep - Register source/dest smart contracts, lock&stake some amounts
+        let source_contract = MockSmartContract::wasm(1 as AccountId);
+        let dest_contract = MockSmartContract::wasm(2 as AccountId);
+        assert_register(1, &source_contract);
+        assert_register(1, &dest_contract);
+
+        let account = 2;
+        let amount = 300;
+        assert_lock(account, amount);
+        assert_stake(account, &source_contract, 100);
+        assert_stake(account, &dest_contract, 100);
+
+        // Advance to B&E subperiod, stake again to check both stakes move
+        advance_to_next_subperiod();
+        assert_stake(account, &source_contract, 100);
+
+        // Move 1 - source -> dest - VOTING CONVERSION 1
+        let move_amount = 150; // reduce bonus status
+        assert_move_stake(account, &source_contract, &dest_contract, move_amount);
+
+        let expected_source_staking_info = SingularStakingInfo {
+            previous_staked: StakeAmount {
+                voting: 50,
+                build_and_earn: 0,
+                era: 2,
+                period: 1,
+            },
+            staked: StakeAmount {
+                voting: 0,
+                build_and_earn: 50,
+                era: 3,
+                period: 1,
+            },
+            bonus_status: 0, // bonus has been forfeited
+        };
+
+        let source_staking_info = StakerInfo::<Test>::get(&account, &source_contract)
+            .expect("Should exist after a successful move operation");
+        assert_eq!(source_staking_info, expected_source_staking_info);
+
+        let expected_dest_staking_info = SingularStakingInfo {
+            previous_staked: StakeAmount {
+                voting: 100,
+                build_and_earn: 0,
+                era: 2,
+                period: 1,
+            },
+            staked: StakeAmount {
+                voting: 100,
+                build_and_earn: 150,
+                era: 3,
+                period: 1,
+            },
+            bonus_status: 1, // bonus is preserved
+        };
+        let dest_staking_info = StakerInfo::<Test>::get(&account, &dest_contract)
+            .expect("Should exist after a successful move operation");
+        assert_eq!(dest_staking_info, expected_dest_staking_info);
+
+        // Move 2 - source -> dest - VOTING CONVERSION 2
+        let move_amount = 50; // full move
+        assert_move_stake(account, &source_contract, &dest_contract, move_amount);
+
+        assert!(StakerInfo::<Test>::get(&account, &source_contract).is_none());
+
+        let expected_dest_staking_info = SingularStakingInfo {
+            previous_staked: StakeAmount {
+                voting: 100,
+                build_and_earn: 150, // TODO: should it not be 0 here?
```

Comment:
Yes, it should remain unchanged.

Reply:
This should be fixed in this commit https://github.com/AstarNetwork/Astar/pull/1418/commits/499b5f448bbd6fb6c62b4fdd72b637f6dba13df1